arrow/arrow.go

```diff
@@ -22,21 +22,23 @@ type ArrowWriter struct {
 // The ArrowWriter will write to filePath.
 // This writing operation is threadsafe.
 func NewArrowIPCFileWriter(f *os.File, fieldNames []string, fieldTypes []arrow.DataType, options ...ipc.Option) (*ArrowWriter, error) {
-	schema := makeSchema(fieldNames, fieldTypes)
+	return NewArrowIPCFileWriterWithSchema(f, makeSchema(fieldNames, fieldTypes), options...)
+}
 
+func NewArrowIPCFileWriterWithSchema(f *os.File, schema *arrow.Schema, options ...ipc.Option) (*ArrowWriter, error) {
 	schemaOption := ipc.WithSchema(schema)
 	writer, err := ipc.NewFileWriter(f, append([]ipc.Option{schemaOption}, options...)...)
 	if err != nil {
 		return nil, err
 	}
 
 	return &ArrowWriter{
-		schema: schema,
+		Schema: schema,
 		writer: writer,
 	}, nil
 }
 
-func (aw *ArrowWriter) writeChunk(record arrow.Record) error {
+func (aw *ArrowWriter) WriteChunk(record arrow.Record) error {
```

Comment:
It still says writeChunk on line 110 and 121

Reply:
Ah those are methods on ArrowRowBuilder, which does have a writeChunk method.